### DISCUSSION TITLE
No Export Entities - I Cannot Find any Entities for Export

### DISCUSSION THREAD

**Discussion by poolitman** [BottlecapDave/HomeAssistant-OctopusEnergy#876] [Q&A] [answered]
Sorry if this has been asked/answered previous but I cannot find anything in discussions or issues. I've also checked the documentation & FAQs.

I can't find any entity in the integration which applies to export or contains export in the entity name, in my HA set up.

I have had the Octopus integration for several months, initially I didn't have my export with Octopus, but 6 weeks ago my export was transferred to Octopus & I received an email from them saying every was set up. I can see my export data in the octopus app.

I've checked the "Enable newly added entities" option is enabled in the systems option & toggled this a number of times.

I've also reloaded the integration a number of times & restarted HA numerous times.

Am I missing something obvious? Is there something else I should do? What else should I check to troubleshoot this?

Thanks

**Comment by Dukeicon** [comment]
I recently had export added to my Octopus account.
I think (but can't remember 100%) that I did the following.

Go to the Octopus intergration, click 'Configure' and then 'Submit'.
Restart HA and wait for the intergration to load.

You should then have an extra device: Electricity MeterExport.

I have 3 in total - Gas Meter - Electricity Meter & Electricity MeterExport.

To get your data into the Energy Dashboard use this guide:
https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/energy_dashboard/#for-electricity_1

You must use the **octopus_energy:** (note the full colon after energy) entity for the stats to appear correctly in the dashboard.

The Octopus Mini doesn't have live Export Data, so any Export Data comes in once a day (mine is around 1-3pm).

**Reply by poolitman** [reply]
Hi

Thank you for the suggestion/s.

I have previously and repeated again today (following your suggestion) going into the octopus integration, clicking "Configure" and then clicking "Submit" and restarting HA. But I still only have 2 devices showing in the integration (electricity meter & gas meter).

I wasn't sure whether or not the export MPAN would show as a separate electric meter (additional device) or just as additional entities. I'm grateful you've confirmed I should have a 3rd device. Perhaps this will help troubleshoot my issue.

I had read that the Octopus Mini doesn't have live export, but I had read than in the octopus app, export showed up as a negative reading on the live screen. Mine doesn't do this, but I do see the export figures on the weekly screen (obviously a day behind).

I hope the above makes sense & someone can suggest what I should do/try next.

**Reply by Dukeicon** [reply]
Yes, I have 3 devices - they are all under the same Octopus account number (I also have a FiT but that is in a separate account).

<img width="242" alt="Screenshot 2024-05-17 at 15 32 39" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/115842708/b034d059-1d98-4e92-83de-368fbb50b901">


Can you see all 3 meters on the Octopus API page?
https://octopus.energy/dashboard/new/accounts/personal-details/api-access

I have 2 Electricity entities (same serial number for both but different MPAN) and 1 Gas entity.
If you can't see both Electricity ones there, then this intergration/HA won't be able to see it either.


I'm trying to remember what I did, but it was about a month ago now and i've been to sleep since then.
The only other thing I can think of is, clicking the 3 dot menu and then reload?

<img width="366" alt="Screenshot 2024-05-17 at 15 30 34" src="https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/115842708/82ae34de-65a7-492f-8a2d-6c71ae442206">


This user here had a similar question/issue and it eventaully started working after refreshing (assume they mean reloading).
https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/discussions/808

**Reply by poolitman** [reply]
Hi

Yes I can see all 3 meters in the Octopus API screen, my electricity export appears first, then electricity import and finally gas.

I have tried reloading the integration, several times, but this didn't work.

The only thing I haven't tried is removing & reinstalling the integration. The reason I haven't tried that is I don't know what will happen i.e. would this be a waste of time.

I've seen the discussion you've mentioned, as you say they just said it started working after they did a refresh, but not what they mean by refresh.

**Reply by BottlecapDave** [reply]
Hello and sorry you're having issues.. The integration will only show entities for meters with an active tariff agreement. 

If you follow the [FAQ](https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/faq/#ive-been-asked-for-my-meter-information-in-a-bug-request-how-do-i-obtain-this) to download diagnostics, this will show all meters the integration can see. You want to look for the meter where `is_export` is `true`. Under this meter you will need an `agreement` with a start date in the past and an end date either `null` or in the future. If you have an agreement that starts in the future, then I'm afraid you'll need to wait until it becomes active before it will appear in the integration. If you do not have this, then you will need to contact OE as it means something in your account isn't quite right.

**Reply by poolitman** [reply]
Hi

I've downloaded the diagnostics and it only shows my electricity import meter & my gas meter. Which, as I replied to Dukeicon (above) is what the integration is showing (2 devices, not 3). 

But the octopus api page (in my octopus account) is showing all 3 meters (electricity export, electricity import & gas). I've double checked, on the octopus api page, that the electric meter serial numbers match & the MPANS are correct, as well as the api key & account number match the ones saved in the integration.

If I need to contact octopus what exactly am I asking them to do/look at? 

If I say the export meter isn't showing in the api I presume they are likely to ask me to check if its showing in my account api page, which it is! Then they are likely to refer me back to the integration authors! 

So any help with the wording of what I'm asking them to check for would be much appreciated.

I don't know if it helps/makes any difference, but I can see export data using the example curl query on the octopus api page, under my electricity export meter.

Thanks

**Reply by BottlecapDave** [reply, marked as answer]
If the meter isn't coming back in the API request, then there is nothing this integration can do as it doesn't know about your export meter. If you contact them via email at [hello@octopus.energy](mailto:hello@octopus.energy?subject=Customer%20API) and state that your export meter isn't coming back in your list of meters when performing the following query

```
query {
	account(accountNumber: "A-XXX") {
		electricityAgreements {
			meterPoint {
				mpan
				meters {
					serialNumber
				}
			}
		}
	}
}
```

You will probably need to provide your account id if you use an address not associated with your account.

Hopefully they'll be able to sort it and you'll be able to see your export meter.

**Reply by poolitman** [reply]
Thanks

I've sent Octopus an email, as suggested and hope they fix this soon.

**Reply by poolitman** [reply]
Thanks

Octopus have sorted this out. They didn't say what they've done, but I now have the export meter appearing in the integration.

**Reply by poolitman** [reply]
Hi

I now have the export meter showing but I don't see an entity to show my actual export in the integration.

The entities which should show power are binary_sensor.octopus_energy_electricity_xxxxx_xxxx_export_off_peak which shows as off even though it is enabled. And the other power entity is sensor.octopus_energy_electricity_xxxx_xxxx_export_previous_accumulative_consumption which appears to be showing a few days worth of export.

Am I doing something wrong?

**Reply by BottlecapDave** [reply]
> The entities which should show power are binary_sensor.octopus_energy_electricity_xxxxx_xxxx_export_off_peak which shows as off even though it is enabled

This only turns on if you are on a tariff for that meter that has 2 or 3 unique rates. If you are on an export tariff with 2 or 3 unique rates, then it should turn on when the lowest rate is active. If this is not happening, please raise a bug report.

>sensor.octopus_energy_electricity_xxxx_xxxx_export_previous_accumulative_consumption which appears to be showing a few days worth of export.

previous consumption sensor is always at least 24 hours out. This is due to the data being delayed via the OE API.

**Reply by poolitman** [reply]
Thanks for the clarification.

So there are no live entities for export other than current rate £ & current standing charge £?

**Reply by BottlecapDave** [reply]
Unfortunately not. The only way to get live data is via an octopus mini, but this doesn't provide any live data for exports other than demand sensor going into the negative.

**Reply by poolitman** [reply]
Thanks once again

I have an Octopus Mini but unfortunately my live data never goes below zero on the Octopus app.

I believe its because of the smart meter (a 118F - Aclara), as my Glow IHD also doesn't show export either (never goes below zero) & there's a reported issue from Glow users who used to receive negative displays when exporting but after an update to their smart meter they lost this facility.

Looks like I'm stuck at the moment.

Thanks
